src/ex2/DomManager.js

```diff
@@ -0,0 +1,201 @@
+        const task_text = document.createElement("span");
+        const task_number = document.createElement("span");
+        const task_buttons_container = document.createElement("div");
+        const task_delete_button = document.createElement("button");
+        const task_complete_button = document.createElement("button");
+
+        // adds class to elements
+        new_task.classList.add("todo_task");
+        new_task.classList.add("animate__animated");
+        new_task.classList.add("animate__fadeIn");
+        task_text.classList.add("task_text");
+        task_number.classList.add("task_number");
+        task_buttons_container.classList.add("task_buttons");
+        task_delete_button.classList.add("delete_task_button");
+        task_delete_button.id = id;
+        task_complete_button.classList.add("complete_task_button");
+
+        // Delete empty state from task container
+        this.task_container.classList.remove("empty");
+        // appends elements to dom
+        this.task_container.appendChild(new_task);
+        task_buttons_container.append(task_delete_button, task_complete_button);
+        new_task.append(task_number, task_text);
+        // image
+        if (Number.isInteger(task_object.id)) {
+            new_task.appendChild(this.PokemonImageElement(task_object));
+            // adds text to task
+            task_text.innerHTML = `Catch ${task_object.name}`;
+        }
+        else if(task_object.id === "Error")
+            task_text.innerHTML = task_object.data;
+        else
+            task_text.innerHTML = task_object;// adds text to task
+        new_task.append(task_buttons_container);
+
+        // adds numbers to tasks
+        task_number.innerText = (this.task_container.children.length).toString().concat(")");
+        this.clearInput();
+
+        // adds icons to buttons
+        task_delete_button.innerHTML = '<i class="fa-solid fa-trash-can"></i>';
+        task_complete_button.innerHTML = '<i class="fa-solid fa-check"></i>';
+
+        // adds event listeners to elements
+        task_delete_button.addEventListener("click", (event) => {
+            delete_call_back(event);
+        });        
+        task_complete_button.addEventListener("click", this.MarkAsCompleteTask);
+        task_text.addEventListener("click", this.TaskClick);
+
+        // updates task counter
+        this.UpdateTaskCounter();
+    }
+    /**
+     * add div if images of pokemon
+     * @param {object} task_object pokemon object
+     * @returns 
+     */
+    PokemonImageElement(task_object) {
+        const pokemon_image_sprites_container = document.createElement("div");
+        pokemon_image_sprites_container.classList.add("image_sprites");
+        Object.entries(task_object.images).forEach((image_attribute, index) => {// adds all images from api
+            if (image_attribute[1] && image_attribute[1] !== Object(image_attribute[1])) {// check if there is an image 
```

Comment:
This condition isn't clear. Did you want to validate that `image_attribute[1]` is not an object?